server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -178,18 +201,11 @@ private boolean checkSegmentFilesIntactWithStartMarker(File localStorageDir)
   }
 
   @Override
-  public Segment getSegment(DataSegment segment, boolean lazy, SegmentLazyLoadFailCallback loadFailed) throws SegmentLoadingException
+  public ReferenceCountingSegment getSegment(DataSegment segment, boolean lazy, SegmentLazyLoadFailCallback loadFailed)
+      throws SegmentLoadingException
   {
-    final ReferenceCountingLock lock = createOrGetLock(segment);
```

Comment:
the lock acquisition has been removed from here since `getSegmentFiles` already does it.